### PR TITLE
Refactor perf module for better readabiity

### PR DESCRIPTION
**Pull request by Czaki**
# References and relevant issues

Fix things pointed in #6575

# Description

Improve readability and remove obsolete code in perf module

### CODE REVIEW

**Review by melonora — CHANGES_REQUESTED** [review]
In `_timer.py`, could you please  improve the readability a bit more by adding information about the different kinds of events, either in the class docstring or as docstring of the `add` functions? Other than that, it is a lot more readable:)

**Review by psobolewskiPhD — COMMENTED** [review]
This looks good, but I agree with @melonora would be nice to have the extra info in the code, even if some of this is documented in the docs:
https://napari.org/stable/howtos/perfmon.html

**Comment by Czaki** [comment]
Could you point exact places where docs should be expanded?

**Comment by psobolewskiPhD** [comment]
I can't make suggestion there easily, but in `PerfTimers`:
docstring for `add_instant_event` currently `"""Add one instant event.`
- would like just few words, 1-line max, what this is/does
docstring for `add_counter_event` currently `"""Add one counter event.`
- would like just few words, 1-line max, what this is/does

I think these are then repeated in DummyTimers

**Comment by Czaki** [comment]
I have updated. 
I do not feel fully familiar with perfmon code. The idea of this PR is to remove bad patterns from code.

**Review by psobolewskiPhD — APPROVED** [review]

File: napari/utils/perf/_timers.py [hunk 1]
```diff
@@ -226,78 +233,82 @@ def block_timer(
...
+    **kwargs: float,
+) -> None:
+    """Add one instant event.
+
+    Parameters
+    ----------
+    name : str
+        Add this event.
+    category : str | None
+        Comma separated categories such has "render,update".
```
Comment (melonora) [inline napari/utils/perf/_timers.py]:
```suggestion
        Comma separated categories such as "render, update".
```

**Review by melonora — APPROVED** [review]
One minor typo but I will pre-approve. Thanks @Czaki !

**Pull request merged**